modulus/utils/filesystem.py

```diff
@@ -69,26 +71,38 @@ def _download_ngc_model_file(path: str, out_path: str, timeout: int = 300) -> st
     # Strip ngc model url prefix
     suffix = "ngc://models/"
     # The regex check
-    pattern = re.compile(f"{suffix}[\w-]+/[\w-]+/[\w-]+@[A-Za-z0-9.]+/[\w/]+")
+    pattern = re.compile(f"{suffix}[\w-]+(/[\w-]+)?/[\w-]+@[A-Za-z0-9.]+/[\w/](.*)")
     if not pattern.match(path):
         raise ValueError(
             "Invalid URL, should be of form ngc://models/<org_id/team_id/model_id>@<version>/<path/in/repo>"
         )
 
     path = path.replace(suffix, "")
-    (org, team, model_version, filename) = path.split("/", 4)
-    (model, version) = model_version.split("@", 1)
+    if len(path.split("@")[0].split("/")) == 3:
+        (org, team, model_version, filename) = path.split("/", 3)
+        (model, version) = model_version.split("@", 1)
+    else:
+        (org, model_version, filename) = path.split("/", 2)
+        (model, version) = model_version.split("@", 1)
+        team = None
+
```

Comment:
Why is it that we make the users submit the path with the model, team (optional), filename, model built in (which they would have to do manually anyways) just for us to then split it apart anyways? Would it not make sense to have the api look something like this:

```python
_download_ngc_model_File(
    org, 
    model_id, 
    version, 
    file_path, 
    team = None, ...
    ):
```

Reply:
This is integrated into the file system, so it makes sense to use a path. Follows hugging face for url structure.